src/webgpu/shader/validation/expression/call/builtin/const_override_validation.ts

```diff
@@ -312,3 +316,74 @@ export function unique<T>(...arrays: Array<readonly T[]>): T[] {
   }
   return [...set];
 }
+
+interface FloatLimits {
+  positive: {
+    max: number;
+  },
+  negative: {
+    min: number;
+  },
+  emax: number;
+}
+
+/**
+ * Provides an easy way to validate steps in an equation that will trigger a validation error with
+ * constant or override values due to overflow/underflow. Typical call pattern is:
+ * 
+ * const vCheck = new ConstantOrOverrideValueChecker(t, Type.f32);
+ * const c = vCheck.checkedResult(a + b);
+ * const d = vCheck.checkedResult(c * c);
+ * const expectedResult = vCheck.allChecksPassed();
+ */
+export class ConstantOrOverrideValueChecker {
+  #allChecksPassed = true;
+  #floatLimits?: FloatLimits;
+  #quantizeFn: QuantizeFunc<number>;
+
+  constructor(private t: ShaderValidationTest, type: ScalarType) {
+    switch (type) {
+      case Type.f32:
+        this.#quantizeFn = quantizeToF32;
+        this.#floatLimits = kValue.f32;
+        break;
+      case Type.f16:
+        this.#quantizeFn = quantizeToF16;
+        this.#floatLimits = kValue.f16;
+        break;
+      default:
+        this.#quantizeFn = (v: number) => v;
+        break;
+    }
+  }
+
+  quantize(value: number): number {
+    return this.#quantizeFn(value);
+  }
+
+  // Some overflow floating point values may fall into an abiguously rounded scenario, where they
+  // can either round up to Infinity or down to the maximum representable value. In these cases the
+  // test should be skipped, because it's valid for implementations to differ.
+  isAmbiguousOverflow(value: number): boolean {
```

Comment:
From reading the code I think that for infinities it will always return false, because abs(+/-Infinit) is Infinity, which will never be less that 2^emax+1.

I think NaN just flows through and returns a false at the end, but I am not 100% sure about that.

PRobably worth being explict in the comment that values greater than 2^emax+1 will round to infinity and thus are not ambigious, i.e. will always fail validation

Reply:
Added a bunch of comments and made the finite check explicit.